ddtrace/tracer/option_test.go

```diff
@@ -98,6 +100,7 @@ func TestStatsdUDPConnect(t *testing.T) {
 }
 
 func TestAutoDetectStatsd(t *testing.T) {
+	internal.TestingWithAgent = true
```

Comment:
Shouldn't this kind of calls have a `defer` statement to revert the global variable to its original value? Also, I think it should be done only in the places where we expect `TestingWithAgent` to be `false` if the default is `true` (although I see that you initialized it to `false` in the `TestMain`).